### PR TITLE
[flang][doc] Added information about `REAL(16)` math intrinsics support.

### PR DESCRIPTION
**Pull request by vzakhari**

### CODE REVIEW

File: flang/docs/Real16MathSupport.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+-->
+
+# Flang support for REAL(16) math intrinsics
+
+To support most `REAL(16)` (i.e. 128-bit float) math intrinsics Flang relies
+on third-party libraries providing the implementation.
+
+`-DFLANG_RUNTIME_F128_MATH_LIB=libquadmath` CMake option can be used
+to build `FortranFloat128Math` library that has unresolved references
+to GCC `libquadmath` library. Flang driver built with this option
```
Comment (psteinfeld) [inline flang/docs/Real16MathSupport.md:16]: This should read "A Flang driver" rather than "Flang driver".

File: flang/docs/Real16MathSupport.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+to any Fortran program. This implies that `libquadmath` library
+has to be available in the standard library paths, so that linker
+can find it. The `libquadmath` library installation into Flang project
+distribution is not automatic in CMake currently.
+
+Testing shows that `libquadmath` versions before GCC-9.3.0 have
+accuracy issues, so it is recommended to distribute the Flang
+package with later versions of `libquadmath`.
+
+Care must be taken by the distributors of Flang package built
```
Comment (psteinfeld) [inline flang/docs/Real16MathSupport.md:27]: This should read "a Flang package" rather than "Flang package"

File: flang/docs/Real16MathSupport.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+with `REAL(16)` support via `libquadmath` because of its licensing
+under GNU Library General Public License. Moreover, static linking
```
Comment (psteinfeld) [inline flang/docs/Real16MathSupport.md:29]: This should read "the GNU" rather than "GNU".

File: flang/docs/Real16MathSupport.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+of `libquadmath` to the Flang users' programs may imply some
+restrictions/requirements. This document is not intended to give
+any legal advice on distributing such a Flang compiler.
+
+Flang compiler targeting systems with `LDBL_MANT_DIG == 113`
+may provide `REAL(16)` math support without `libquadmath`
```
Comment (psteinfeld) [inline flang/docs/Real16MathSupport.md:35]: This should read "without a" rather than "without".

File: flang/docs/Real16MathSupport.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+dependency, using standard `libc` APIs for `long double`
```
Comment (psteinfeld) [inline flang/docs/Real16MathSupport.md:36]: This should read "for the" rather than "for".

**Review by psteinfeld — APPROVED** [review]
Thanks for writing this up!

Feel free to merge after deciding on my suggested changes.

**Comment by vzakhari** [comment]
Thank you for the review, Pete! I will apply your suggestions!

**Pull request merged**